evennia/commands/default/building.py

```diff
@@ -1515,31 +1518,130 @@ def check_attr(self, obj, attr_name):
         """
         return attr_name
 
+    def split_nested_attr(self, attr):
+        """
+        Yields tuples of (possible attr name, nested keys on that attr).
+        For performance, this is biased to the deepest match, but allows compatability
+        with older attrs that might have been named with `[]`'s.
+
+        > list(split_nested_attr("nested['asdf'][0]"))
+        [
+            ('nested', ['asdf', 0]),
+            ("nested['asdf']", [0]),
+            ("nested['asdf'][0]", []),
+        ]
+        """
+        quotes = '"\''
+
+        def clean_key(val):
+            val = val.strip('[]')
+            if val[0] in quotes:
+                return val.strip(quotes)
+            if val[0] == LIST_APPEND_CHAR:
+                # List insert/append syntax
+                return val
+            try:
+                return int(val)
+            except ValueError:
+                return val
+
+        parts = self.nested_re.findall(attr)
+
+        base_attr = ''
+        if parts:
+            base_attr = attr[:attr.find(parts[0])]
+        for index, part in enumerate(parts):
+            yield (base_attr, [clean_key(p) for p in parts[index:]])
+            base_attr += part
+        yield (attr, [])
+
+    def do_nested_lookup(self, value, *keys):
+        result = value
+        for key in keys:
+            try:
+                result = result.__getitem__(key)
+            except (IndexError, KeyError, TypeError):
+                return self.not_found
+        return result
+
     def view_attr(self, obj, attr):
         """
         Look up the value of an attribute and return a string displaying it.
         """
-        if obj.attributes.has(attr):
-            return "\nAttribute %s/%s = %s" % (obj.name, attr,
-                                               obj.attributes.get(attr))
+        for key, nested_keys in self.split_nested_attr(attr):
```

Comment:
If doing something like `set obj/existing_nested_attr['non_existing']`, the `else` clause below leads to a somewhat confusing error `x has no Attribute y`. At this point it is clear that the user _has_ the Attribute - the error is rather than it does not contain the nested key/index. It would be better to distinguish these two error cases.

Reply:
OK, I added a note if a nested lookup was attempted.